ledger/participant-integration-api/src/test/suite/scala/platform/usermanagement/CachedUserManagementStoreSpec.scala

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.usermanagement
+
+import com.daml.ledger.api.domain.{User, UserRight}
+import com.daml.ledger.participant.state.index.impl.inmemory.InMemoryUserManagementStore
+import com.daml.ledger.participant.state.index.v2.UserManagementStore
+import com.daml.ledger.participant.state.index.v2.UserManagementStore.UserNotFound
+import com.daml.ledger.resources.TestResourceContext
+import com.daml.lf.data.Ref
+import org.mockito.{ArgumentMatchersSugar, MockitoSugar}
+import org.scalatest.freespec.AsyncFreeSpec
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.Future
+
+// TODO participant user management: Maybe using a custom dummy (or a spy on InMemory version) rather than Mockito.mock would result in a simpler test
+class CachedUserManagementStoreSpec
```

Comment:
This one could be combined with the general `UserManagementStoreSpec`; i.e., the unit tests for `UserManagementStore` could and likely should be executed both against the `InMemoryUserManagementStore` and `CachedUserManagementStore(InMemoryUserManagementStore)`.